hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/cache/CacheModule.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.mono.cache;
+
+import com.hedera.node.app.service.mono.context.properties.GlobalDynamicProperties;
+import com.hedera.node.app.service.mono.state.StateModule;
+import com.hedera.node.app.service.mono.state.migration.AccountStorageAdapter;
+import com.hedera.node.app.service.mono.state.migration.TokenRelStorageAdapter;
+import com.hedera.node.app.service.mono.state.migration.UniqueTokenMapAdapter;
+import dagger.Module;
+import dagger.Provides;
+import java.util.function.Supplier;
+
+@Module(includes = {StateModule.class})
+public interface CacheModule {
+    @Provides
+    static EntityMapWarmer mapWarmer(
```

Comment:
I'd prefer to let Dagger handle this via an `@Inject`-annotated constructor.

Reply:
I would've preferred to use Dagger, but wasn't quite able to figure it out – it kept instantiating multiple instances. I'll try again.